src/app/dataValidator/dataValidator.js

```diff
@@ -10,16 +10,54 @@ const yamlSchema = fs.readFileSync('./././data/schema.yaml', 'utf8');
 const { components } = yaml.load(yamlSchema);
 const { schemas } = components;
 
-const validateNode = (currentSchemaNode, dataNode) => {
-  validateType(currentSchemaNode.type, dataNode);
+const validateNode = (currentSchemaNode, dataNode, schemaName) => {
+  validateType(currentSchemaNode.type, dataNode, schemaName);
 
   if (currentSchemaNode.enum) {
-    validateEnum(currentSchemaNode.enum, dataNode);
+    validateEnum(currentSchemaNode.enum, dataNode, schemaName);
   }
 
   if (currentSchemaNode.required) {
-    validateRequired(currentSchemaNode.required, dataNode);
+    validateRequired(currentSchemaNode.required, dataNode, schemaName);
   }
+
+  if (currentSchemaNode.properties) {
+    validateProperties(currentSchemaNode, dataNode, schemaName); // eslint-disable-line no-use-before-define
+  }
+};
+
+const isRequiredProperty = (property, schema) => {
```

Comment:
There are tiny files such as dataValidator/validateEnum.js factored out, but here they're left in the 'main' file. There needs to be a clear distinction/reason for separation. Am I missing the reason or does it need tidying?

Reply:
This has now be separated out